### PR TITLE
Start removal of gfx_draw_string_*()

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/drawing/Text.h [hunk 1]
```diff
@@ -48,3 +49,16 @@ class StaticLayout
...
+
+void DrawTextBasic(
+    rct_drawpixelinfo* dpi, const ScreenCoordsXY& coords, rct_string_id format, const Formatter& ft, uint8_t colour,
+    TextAlignment alignment, bool underline = false);
+void DrawTextBasic(
+    rct_drawpixelinfo* dpi, const ScreenCoordsXY& coords, rct_string_id format, const void* args, uint8_t colour,
+    TextAlignment alignment, bool underline = false);
+void DrawTextEllipsised(
+    rct_drawpixelinfo* dpi, const ScreenCoordsXY& coords, int32_t width, rct_string_id format, const Formatter& ft,
+    uint8_t colour, TextAlignment alignment, bool underline = false);
```
Comment (Gymnasiast) [inline src/openrct2/drawing/Text.h:61]: `colour_t`

File: src/openrct2/drawing/Drawing.h [hunk 2]
```diff
@@ -648,24 +649,23 @@ void FASTCALL gfx_draw_sprite_raw_masked_software(
 // string
 void gfx_draw_string(rct_drawpixelinfo* dpi, const_utf8string buffer, uint8_t colour, const ScreenCoordsXY& coords);
 
+/** @deprecated */
```
Comment (ZehMatt) [inline src/openrct2/drawing/Drawing.h:652]: You could use https://en.cppreference.com/w/cpp/language/attributes/deprecated, this will add some compiler warnings but will actually make it clear that its not supposed to be used anymore.

Comment (Gymnasiast) [reply src/openrct2/drawing/Drawing.h:652]: The problem with that is that compilation will fail if a compiler warning is output in our project.

Comment (ZehMatt) [reply src/openrct2/drawing/Drawing.h:652]: Is there no way to ignore this specific warning maybe?

Comment (Gymnasiast) [reply src/openrct2/drawing/Drawing.h:652]: I don't know, but I'm sure @janisozaur does.

Comment (ZehMatt) [reply src/openrct2/drawing/Drawing.h:652]: I mean its not that big of a deal, we are going to remove them anyway soonish I guess.

File: src/openrct2/localisation/Localisation.h [hunk 3]
```diff
@@ -112,6 +112,11 @@ class Formatter
         return CurrentBuf;
     }
 
+    auto GetStartBuf() const
```
Comment (ZehMatt) [inline src/openrct2/localisation/Localisation.h:115]: Unrelated?

**Review by ZehMatt — APPROVED** [review]
Besides the one unrelated change, lgtm.

File: src/openrct2/localisation/Localisation.h (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2/localisation/Localisation.h:115]: No. Take a look at `DrawTextBasic` and `DrawTextEllipsised`. They need to get the start buffer. If I just call `Buf()`, it will crash the game.

**Comment by duncanspumpkin** [comment]
Why are we exposing the args overload? Nothing should be using that version of the function. Everything should use the formatter.
